test/Microsoft.Extensions.Primitives.Tests/CompositeChangeTokenTest.cs

```diff
@@ -0,0 +1,175 @@
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken, thirdChangeToken });
+            var count1 = 0;
+            var count2 = 0;
+
+            compositeChangeToken.RegisterChangeCallback(_ => count1++, null);
+            compositeChangeToken.RegisterChangeCallback(_ => count2++, null);
+
+            // Act
+            firstChangeToken.RaiseCallback(null);
+            secondChangeToken.RaiseCallback(null);
+            thirdChangeToken.RaiseCallback(null);
+
+            // Assert
+            Assert.Equal(1, count1);
+            Assert.Equal(1, count2);
+        }
+
+        [Fact]
+        public void HasChanged_IsTrue_IfAnyTokenHasChanged()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken { HasChanged = true };
+            var thirdChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken, thirdChangeToken });
+
+            // Act & Assert
+            Assert.True(compositeChangeToken.HasChanged);
+        }
+
+        [Fact]
+        public void HasChanged_IsFalse_IfNoTokenHasChanged()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken });
+
+            // Act & Assert
+            Assert.False(compositeChangeToken.HasChanged);
+        }
+
+        [Fact]
+        public void ActiveChangeCallbacks_IsTrue_IfAnyTokenHasActiveChangeCallbacks()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken { ActiveChangeCallbacks = true };
+            var thirdChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken, thirdChangeToken });
+
+            // Act & Assert
+            Assert.True(compositeChangeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public void ActiveChangeCallbacks_IsFalse_IfNoTokenHasActiveChangeCallbacks()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken });
+
+            // Act & Assert
+            Assert.False(compositeChangeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public async Task CallbackRaisedOnce_ConcurrentThreadsBlocked()
+        {
+            // Arrange
+            var event1 = new AutoResetEvent(false);
```

Comment:
I chose `AutoResetEvent` because they automatically close after releasing one thread at a time. Any reason why `ManualResetEvent` is preferred?